### ISSUE TITLE
Errors in Phantom JS 2.0.1 with 0.7.7

### ISSUE THREAD

**Issue by jherdman** [machty/ember-concurrency#59]
I'm attempting an upgrade to 0.7.7 so that I no longer need to include the babel polyfill. I have a number of errors when running tests with Phantom 2.0.1 though:

```
not ok 478 PhantomJS 2.0 - Unit | Service | membership freshness watcher: stops polling if a server responds with an error
    ---
        actual: >
            null
        stack: >
                at defaultSubject (http://localhost:7000/assets/test-support.js:14287:28)
                at http://localhost:7000/assets/test-support.js:14319:41
                at http://localhost:7000/assets/tests.js:12665:31
                at wrapper (http://localhost:7000/assets/test-support.js:13086:34)
                at runTest (http://localhost:7000/assets/test-support.js:2796:32)
                at run (http://localhost:7000/assets/test-support.js:2781:11)
                at http://localhost:7000/assets/test-support.js:2923:14
                at process (http://localhost:7000/assets/test-support.js:2582:24)
                at begin (http://localhost:7000/assets/test-support.js:2564:9)
                at http://localhost:7000/assets/test-support.js:2624:9
        message: >
            Died on test #1     at testWrapper (http://localhost:7000/assets/test-support.js:13115:16)
                at test (http://localhost:7000/assets/test-support.js:13128:44)
                at http://localhost:7000/assets/tests.js:12641:24
                at exports (http://localhost:7000/assets/vendor.js:127:37)
                at requireModule (http://localhost:7000/assets/vendor.js:30:25)
                at require (http://localhost:7000/assets/test-loader.js:67:16)
                at loadModules (http://localhost:7000/assets/test-loader.js:58:25)
                at load (http://localhost:7000/assets/test-loader.js:89:35)
                at http://localhost:7000/assets/test-support.js:6477:20: undefined is not a constructor (evaluating 'factory.create(options)')
```

Additionally:

```
not ok 477 PhantomJS 2.0 - Unit | Service | membership freshness watcher: polls for new content
    ---
        actual: >
            null
        stack: >
                at http://localhost:7000/assets/student-app.js:9824:39
                at exports (http://localhost:7000/assets/vendor.js:127:37)
                at requireModule (http://localhost:7000/assets/vendor.js:30:25)
                at _extractDefaultExport (http://localhost:7000/assets/vendor.js:168998:27)
                at resolveOther (http://localhost:7000/assets/vendor.js:168733:53)
                at superWrapper (http://localhost:7000/assets/vendor.js:33912:26)
                at resolve (http://localhost:7000/assets/vendor.js:16672:47)
                at _setupIsolatedContainer (http://localhost:7000/assets/test-support.js:14387:64)
                at setupContainer (http://localhost:7000/assets/test-support.js:14222:37)
                at http://localhost:7000/assets/test-support.js:13294:30
                at initializePromise (http://localhost:7000/assets/vendor.js:64658:15)
                at Promise (http://localhost:7000/assets/vendor.js:66510:38)
                at nextStep (http://localhost:7000/assets/test-support.js:13293:52)
                at invokeSteps (http://localhost:7000/assets/test-support.js:13300:22)
                at setup (http://localhost:7000/assets/test-support.js:13250:30)
                at setup (http://localhost:7000/assets/test-support.js:13055:28)
                at callHook (http://localhost:7000/assets/test-support.js:2822:24)
                at runHook (http://localhost:7000/assets/test-support.js:2815:13)
                at process (http://localhost:7000/assets/test-support.js:2582:24)
                at begin (http://localhost:7000/assets/test-support.js:2564:9)
                at http://localhost:7000/assets/test-support.js:2624:9
        message: >
            Promise rejected before polls for new content: undefined is not a function (evaluating '(0, _emberConcurrency.task)')
        Log: |
```

These tests pass in Chrome Canary (53.0.2767.0 canary (64-bit), and Firefox 49.

**Comment by machty** [comment]
Do the tests pass if you add `includePolyfill:true` back in?

**Comment by jherdman** [comment]
They do. There's also a deprecation warning that I didn't include above that is now gone as well. For the sake of completeness, here it is:

```
              text: '\'DEPRECATION: In Ember 2.0 service factories must have an `isServiceFactory` property set to true. You registered [object Object] as a service factory. Either add the `isServiceFactory` property to this factory or extend from Ember.Service. [deprecation id: ember-application.validate-type]\\n    \\tat raiseOnDeprecation (http://localhost:7000/assets/vendor.js:17063:17)\\n    \\tat http://localhost:7000/assets/vendor.js:17155:15\\n    \\tat invoke (http://localhost:7000/assets/vendor.js:17171:21)\\n    \\tat deprecate (http://localhost:7000/assets/vendor.js:17124:37)\\n    \\tat deprecate (http://localhost:7000/assets/vendor.js:27163:42)\\n    \\tat validateType (http://localhost:7000/assets/vendor.js:16984:33)\\n    \\tat resolve (http://localhost:7000/assets/vendor.js:16679:51)\\n    \\tat _setupIsolatedContainer (http://localhost:7000/assets/test-support.js:14387:64)\\n    \\tat setupContainer (http://localhost:7000/assets/test-support.js:14222:37)\\n    \\tat http://localhost:7000/assets/test-support.js:13294:30\\n    \\tat initializePromise (http://localhost:7000/assets/vendor.js:64658:15)\\n    \\tat Promise (http://localhost:7000/assets/vendor.js:66510:38)\\n    \\tat nextStep (http://localhost:7000/assets/test-support.js:13293:52)\\n    \\tat invokeSteps (http://localhost:7000/assets/test-support.js:13300:22)\\n    \\tat setup (http://localhost:7000/assets/test-support.js:13250:30)\\n    \\tat setup (http://localhost:7000/assets/test-support.js:13055:28)\\n    \\tat callHook (http://localhost:7000/assets/test-support.js:2822:24)\\n    \\tat runHook (http://localhost:7000/assets/test-support.js:2815:13)\\n    \\tat process (http://localhost:7000/assets/test-support.js:2582:24)\\n    \\tat begin (http://localhost:7000/assets/test-support.js:2564:9)\\n    \\tat http://localhost:7000/assets/test-support.js:2624:9\'\n' }
```

**Comment by machty** [comment]
Can you try running with

```
    "ember-concurrency": "git://github.com/machty/ember-concurrency#a327c30b0064b430151be97c3d55929c8055dfe8",
```

in your package.json and lemme know if the tests pass?

**Comment by jherdman** [comment]
Worked like a charm!

**Comment by machty** [comment]
Fixed on on ember-concurrency#0.7.8

**Comment by jherdman** [comment]
Many thanks! ❤️
